### ISSUE TITLE
序列分类和奖励模型训练，分类头随机初始化不可控制

### ISSUE THREAD

**Issue by vibe-viscot** [modelscope/ms-swift#2937]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
序列分类和奖励模型训练，分类头随机初始化不可控制，导致训练结果无法复现。
我刚才什么都没改，跑了一下多模态序列分类https://github.com/modelscope/ms-swift/blob/main/examples/train/seq_cls/qwen2_vl/sft.sh。
loss在同样的一套超参数下，每次训练都变化很大，导致训练完的结果完全不一样。

第一次训练的第一个step：
{'loss': 14.734375, 'acc': 0.0, 'grad_norm': 125.29614258, 'learning_rate': 1.61e-06, 'memory(GiB)': 10.85, 'train_speed(iter/s)': 0.117463, 'epoch': 0.0, 'global_step/max_steps': '1/1237', 'percentage': '0.08%', 'elapsed_time': '8s', 'remaining_time': '2h 52m 7s'}

第二次训练的第一个step：
{'loss': 10.69628906, 'acc': 0.0, 'grad_norm': 82.68572235, 'learning_rate': 1.61e-06, 'memory(GiB)': 10.85, 'train_speed(iter/s)': 0.118098, 'epoch': 0.0, 'global_step/max_steps': '1/1237', 'percentage': '0.08%', 'elapsed_time': '8s', 'remaining_time': '2h 51m 18s'}

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
CUDA12.4，Ubuntu2204，python3.9，4070，torch2.5.1

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Jintao-Huang** [comment]
你是希望相同的seed，有类似的初始loss嘛

**Comment by vibe-viscot** [comment]
嗯

**Comment by vibe-viscot** [comment]
当时解决了呀，反正我单卡跑起来没问题了，你是多卡跑的吗？还是说你的swift版本不是最新的？

**Comment by Jintao-Huang** [comment]
应该已经修复了 试试升级一下ms-swift

**Comment by zhuchenxi** [comment]
用了最新的，多卡差异还是很大。单卡简单测了下，初始epoch还是有差异。

**Comment by Jintao-Huang** [comment]
swift版本是什么

**Comment by Sakura-not-sleep** [comment]
请问这个问题有解决吗，我也是无法复现结果，每次都不一样，3.2.0的swift版本
